### PR TITLE
BOAC-1055, read-only cohort naming is whack-a-mole

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-1055
https://jira.ets.berkeley.edu/jira/browse/BOAC-1054

### CODE REVIEW

File: boac/api/util.py [hunk 1]
```diff
@@ -85,15 +85,18 @@ def decorate_cohort(
...
+    cohort_name = 'My Students' if is_read_only and current_user.uid == criteria['advisorLdapUid'] else cohort.label
     decorated = {
         'id': cohort.id,
         'code': cohort.id,
-        'isReadOnly': is_read_only_cohort(cohort),
-        'label': cohort.label,
-        'name': cohort.label,
+        'isReadOnly': is_read_only,
+        'label': cohort_name,
+        'name': cohort_name,
```
Comment (johncrossman) [inline boac/api/util.py]: I created https://jira.ets.berkeley.edu/jira/browse/BOAC-1056 to rename `label` column in db.

File: boac/api/util.py [hunk 2]
```diff
@@ -85,15 +85,18 @@ def decorate_cohort(
     include_profiles=False,
     include_alerts_for_uid=None,
 ):
+    criteria = cohort if isinstance(cohort.filter_criteria, dict) else json.loads(cohort.filter_criteria)
+    is_read_only = is_read_only_cohort(cohort)
+    # In odd circumstances we override the cohort's actual name
+    cohort_name = 'My Students' if is_read_only and current_user.uid == criteria['advisorLdapUid'] else cohort.label
```
Comment (paulkerschen) [inline boac/api/util.py]: @johncrossman Will this throw a KeyError if advisorLdapUid isn't among the criteria?

Comment (johncrossman) [reply boac/api/util.py]: Good catch. It might.

Comment (johncrossman) [reply boac/api/util.py]: Amendment is in.

**Comment by paulkerschen** [comment]
The Mole reached out from under his blanket, felt for the Rat’s paw in the darkness, and gave it a squeeze. ‘I’ll do whatever you like, Ratty,’ he whispered. ‘Shall we run away to-morrow morning, quite early—VERY early—and go back to our dear old hole on the river?’

**Pull request merged**